src/usdb_syncer/notes_parser.py

```diff
@@ -438,20 +453,104 @@ def minimum_song_length(self) -> str:
 
         return f"{minutes:02.0f}:{seconds:02.0f}"
 
-    def set_first_timestamp_to_zero(self) -> None:
+    def fix_first_timestamp(self) -> None:
         """Shifts all notes such that the first note starts at beat zero and adjusts
         GAP accordingly
         """
         if (offset := self.notes.start()) == 0:
+            # round GAP to nearest 10 ms
+            self.headers.gap = round(self.headers.gap / 10) * 10
             return
         for line in self.notes.all_lines():
             line.shift(-offset)
-        self.headers.gap = self.headers.gap + beats_to_ms(offset, self.headers.bpm)
+        self.headers.gap = (
+            round((self.headers.gap + beats_to_ms(offset, self.headers.bpm)) / 10) * 10
+        )
+
+    def fix_low_bpm(self) -> None:
+        """(repeatedly) multiplies BPM value and all note timings by two
+        until the BPM is in a range (200, 400)"""
+        bpm_threshold = 200.0
+
+        if (original_bpm := self.headers.bpm) >= bpm_threshold:
+            return
+
+        while self.headers.bpm < bpm_threshold:
+            self.headers.bpm *= 2
+
+        timing_factor = int(self.headers.bpm / original_bpm)
```

Comment:
Maybe more elegantly:
```py
import math

# how often to multiply bpm until it is larger or equal to the threshold
exp = math.ceil(math.log2(bpm_threshold / self.headers.bpm))
factor = 2**exp
```

Reply:
In fact I had that in mind, but then was unsure if I should import math for that when a simple while-loop does the job =) It is now using math’s ceil() and log2() expression.